Sources/Commands/Utilities/PluginDelegate.swift

```diff
@@ -383,14 +384,38 @@ final class PluginDelegate: PluginInvocationDelegate {
         // Create a build system for building the target., skipping the the cache because we need the build plan.
         let buildSystem = try swiftCommandState.createBuildSystem(explicitBuildSystem: .native, cacheBuildManifest: false)
 
-        // Find the target in the build operation's package graph; it's an error if we don't find it.
+        // Find the target in the build operation's package graph
+        // 1. First look for a matching destination module
+        // 2. If not found, search for a matching tools module, only allowing
+        //    modules that should be built for the host (macro, plugin, test).
+        // 3. Error if no matching targets are found.
+        //
+        // FIXME: dynamic graph
+        // This should work by requesting a build for a target w/ destination:
+        // .destination and generating the graph needed on demand instead of
+        // looking for a preexisting matching target.
```

Comment:
Maybe we should add a `func buildsetFor(target: String) throws -> Buildset` to `BuildSystem` protocol which would encapsulate most of the logic here and allow you to select appropriate build parameters based on resulting buildset.

Reply:
I think the client should be able to request any module built for the target/destination platform and we should create the graph that does that. Is that what you're suggesting here?

Reply:
Yes, client would pass a name and we'd start with `.destination` and if that doesn't exist produce a `.host` if it's appropriate.